crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -270,3 +292,35 @@ impl CachedRenderPipelinePhaseItem for AlphaMask3dPrepass {
         self.key.pipeline
     }
 }
+
+pub fn prepass_target_descriptors(
+    normal_prepass: bool,
+    motion_vector_prepass: bool,
+    deferred_prepass: bool,
+) -> Vec<Option<ColorTargetState>> {
+    vec![
```

Comment:
It's hard not to make a 

```rust
let foo = |format| ColorTargetState {
  format,
  // BlendState::REPLACE is not needed here, and None will be potentially much faster in some cases.
  blend: None,
  write_mask: ColorWrites::ALL
};
```

which would lead to

```rust
vec![
  normal_prepass.then_some(foo(NORMAL_PREPASS_FORMAT)),
  motion_vector_prepass.then_some(foo(MOTION_VECTOR_PREPASS_FORMAT)),
  ..
]
```

but arguably what you already have is more readable (less is more), I'm always on the fence about these.

I'd like to know your preference in case it helps me decide in my own coding in the future?

Reply:
This code was written by @pcwalton.

I would keep it like this myself. No need to add closures and indirection for so little code, just makes it harder to understand what is happening.